torch/csrc/jit/constants.cpp

```diff
@@ -140,117 +140,63 @@ c10::optional<Value*> tryInsertConstant(
-          } else if (
-              type->cast<TupleType>() &&
-              node->kindOf(attr::value) == AttributeKind::ival) {
-            const auto& tup = node->ival(attr::value);
-            TORCH_INTERNAL_ASSERT(tup.isTuple());
-            return [tup](Stack& stack) {
-              push(stack, tup);
-              return 0;
-            };
-          } else if (type->isSubtypeOf(ListType::ofInts())) {
-            const auto& is = node->is(attr::value);
-            return [is](Stack& stack) {
-              push(stack, is);
-              return 0;
-            };
-          } else if (type->isSubtypeOf(ListType::ofFloats())) {
-            const auto& fs = node->fs(attr::value);
-            return [fs](Stack& stack) {
-              push(stack, fs);
-              return 0;
-            };
-          } else if (type->isSubtypeOf(ListType::ofBools())) {
-            const auto bs = fmap<bool>(node->is(attr::value));
-            return [bs](Stack& stack) {
-              push(stack, bs);
-              return 0;
-            };
-          } else if (type->isSubtypeOf(ListType::ofTensors())) {
-            const auto& ts = node->ts(attr::value);
-            return [ts](Stack& stack) {
-              push(stack, ts);
-              return 0;
-            };
-          } else if (type->isSubtypeOf(ListType::ofStrings())) {
-            const auto& ss = node->ss(attr::value);
-            auto vals = c10::impl::GenericList(StringType::get());
-            for (const auto& str : ss) {
-              vals.push_back(str);
-            }
-            return [vals](Stack& stack) {
-              push(stack, vals);
-              return 0;
-            };
-          } else if (type == StringType::get()) {
-            const auto& s = node->s(attr::value);
-            return [s](Stack& stack) {
-              push(stack, s);
-              return 0;
-            };
-          } else if (type == DeviceObjType::get()) {
-            auto d = c10::Device(node->s(attr::value));
-            return [d](Stack& stack) {
-              push(stack, d);
-              return 0;
-            };
-          } else if (node->mustBeNone()) {
-            return [](Stack& stack) {
-              push(stack, IValue());
-              return 0;
-            };
-          } else {
-            std::stringstream ss;
-            ss << "constant literal not supported for: " << type->str();
-            throw std::runtime_error(ss.str());
-          }
-        },
-        aliasAnalysisInternalSpecialCase()),
-});
-
 c10::optional<IValue> toIValue(const Value* v) {
   if (v->node()->kind() != prim::Constant || v->type()->cast<FunctionType>()) {
     return c10::nullopt;
   }
-  // use implementation of prim::Constant to compute the output IValue
-  auto op = v->node()->getOperation();
-  Stack stack;
-  op(stack);
-  return stack.back();
+  const Node* node = v->node();
+  const TypePtr& type = v->type();
+  if (type->isSubtypeOf(TensorType::get())) {
```

Comment:
Shouldn't we at least check that `node->kind() == prim::Constant` before we start blindly reaching for its attributes?